Explain the insecure by default nature of secrets (#24930)
kubernetes/website: content/en/docs/concepts/configuration/secret.md

Pod specification or in an image. Users can create Secrets and the system
also creates some Secrets.

> Kubernetes Secrets are, by default, stored as unencrypted Base64 strings. Once written

Reviewer: ```suggestion
Kubernetes Secrets are, by default, stored as unencrypted base64-encoded strings. Once written
```

> to Kubernetes, they can be retrieved - as plain text - by anyone with API access. In order

Reviewer: ```suggestion
to Kubernetes, they can be retrieved - as plain text - by anyone with API access, **or by anyone with access to Kubernetes' underlying data store, etcd**. In order
```

Reviewer 2: If someone has enabled [encryption at rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) - which I hope they would - and are protecting the encryption keys appropriately, then access to etcd _doesn't_ get you access to Secrets in a useful form.

Reviewer: While true, that's not the _default_ behavior

Reviewer 2: Let's leave readers aware of both the risks of the defaults and the opportunities to address that. It's only correct to document that “anyone with access to Kubernetes' underlying data store” can read Secrets if we can qualify that statement.

Reviewer: Updated suggestion.

Reviewer 2: Thanks @sethvargo

> to safely deploy Secrets, we recommend you (at a minimum):

Reviewer 3: ```suggestion
to safely use Secrets, we recommend you (at a minimum):
```

> 1. [Turn on Encryption at Rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/)

Reviewer 3: ```suggestion
1. [Enable Encryption at Rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) for Secrets
```

Reviewer 2: ```suggestion
1. [Enable Encryption at Rest](/docs/tasks/administer-cluster/encrypt-data/) for Secrets
```
(relative URL)

> 2. [Enable RBAC rules that restrict reading and writing the secret](https://kubernetes.io/docs/reference/access-authn-authz/authorization/)

Reviewer 4: Secrets have a special case when it comes to "restrict reading and writing the Secret," as Secrets may also be obtained implicitly by anyone with the permission to create a Pod. Probably worth noting.

Author: How about this?

```
2. [Enable RBAC rules that restrict reading and writing the secret]. Be aware that secrets can be obtained implicitly by anyone with the permission to create a Pod.
```

> 2. [Enable RBAC rules that restrict reading and writing the secret](https://kubernetes.io/docs/reference/access-authn-authz/authorization/)

Reviewer 3: ```suggestion
2. [Enable RBAC rules that restrict reading and writing the Secret](https://kubernetes.io/docs/reference/access-authn-authz/authorization/)
```

Reviewer 2: ```suggestion
2. [Enable RBAC rules that restrict reading and writing the Secret](/docs/reference/access-authn-authz/authorization/)
```
(relative URL)
Does also assume the reader is using RBAC - a fairly safe assumption though

Pod specification or in an image. Users can create Secrets and the system
also creates some Secrets.

{{< caution >}}
Kubernetes Secrets are, by default, stored as unencrypted base64-encoded

> strings. By default, once they are written to Kubernetes, they can be retrieved

Reviewer 5: nit: line 29, Does it make sense to remove "once they are written to Kubernetes"?
[Once written], Secrets can be retrieved as plain text by anyone with API access or anyone with access ...

access, or anyone with access to Kubernetes' underlying data store, etcd. In
order to safely use Secrets, we recommend you (at a minimum):

> 1. [Enable Encryption at Rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) for Secrets.

Reviewer 5: Hi @adamhjk . Noted a couple of style nits, otherwise, looks OK.
nit: avoid "we".  Who recommends? You could write something like,
`To safely use Secrets, it is recommended that you ...`
nit: Use intra-site links:
```
[Enable Encryption at Rest](/docs/tasks/administer-cluster/encrypt-data/)
[Enable RBAC rules that restrict reading and writing the Secret](/docs/reference/access-authn-authz/authorization/)
```

Reviewer 6: I agree, other than that the commit looks great.

Reviewer 2: ```suggestion
1. [Enable Encryption at Rest](/docs/tasks/administer-cluster/encrypt-data/) for Secrets.
```

> 2. [Enable RBAC rules that restrict reading and writing the Secret](https://kubernetes.io/docs/reference/access-authn-authz/authorization/). Be aware that secrets can be obtained implicitly by anyone with the permission to create a Pod.

Reviewer 2: ```suggestion
2. [Enable or configure RBAC rules](/docs/reference/access-authn-authz/authorization/) that restrict reading and writing the Secret. Be aware that secrets can be obtained implicitly by anyone with the permission to create a Pod.
```

> 2. [Enable RBAC rules that restrict reading and writing the Secret](https://kubernetes.io/docs/reference/access-authn-authz/authorization/) Be aware that secrets can be obtained implicitly by anyone with the permission to create a Pod.

Reviewer 5: nit: Add period before "Be aware that ..." ?
